sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -1196,6 +1196,24 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSparkSession
       }
     }
   }
+
+  test("SPARK-34167: read LongDecimals with precision < 10, VectorizedReader off") {
```

Comment:
BTW, is there a way for Spark to generate these file?

Reply:
As far as I know no, Spark never writes Decimal with precision < 10 as LongDecimal.